### PR TITLE
Feature/extend

### PR DESCRIPTION
**Pull request by kylebaron**
# Summary
- Adds `SETUP__:` field to name an extension yspec file
- trigger extension with this file on call to `ys_load()`
- provided public function for extending a spec not named in `SETUP__:`
- New field is set (`extended_from`) when columns are external to the primary spec
- Refactor `summary.yspec` method to show the extension yaml file under `source`
- Refactor `summary.yspec` to list info on each line, including extension status
- Update dependency versions

### CODE REVIEW

File: R/extend.R [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+#' modeling outputs (e.g. `IPRED`).
+#' 
+#' @details
+#' The extension `yspec` object cannot contain any columns in common with the 
+#' primary `yspec` object. 
+#' 
+#' @param x A `yspec` object. 
+#' @param file The path to a yaml specification file to load and join to `x`; 
+#' if `file` is not passed, the `yspec` object will be searched for the
+#' `extend_file` attribute in `SETUP__:` and will fail if it is not fould.
```
Comment (KatherineKayMRG) [inline R/extend.R:16]: typo in
```suggestion
#' `extend_file` attribute in `SETUP__:` and will fail if it is not found.
```

**Review by KatherineKayMRG — APPROVED** [review]
I'm probably not the right person to review the code for this one but conceptually, and based on your comments, it looks like it's doing what we want. Other than a typo in the comments this looks good to me.

**Review by timwaterhouse — APPROVED** [review]
Looks good. I just wonder if you'd meant to add another test with `ext_bad` for the situation where a column already exists in the primary spec.

**Comment by kylebaron** [comment]
Thanks, @timwaterhouse ; I originally had that erroring if there were common columns; but I realized we had this `ys_join` function that would just drop common cols on the right; will refactor that test to check for that behavior. Thanks for picking that up.

**Pull request merged**